app/screens/AppNavigator/screens/Balances/ConvertScreen.tsx

```diff
@@ -0,0 +1,184 @@
+import { CTransactionSegWit, TransactionSegWit } from '@defichain/jellyfish-transaction'
+import { AddressToken } from '@defichain/whale-api-client/dist/api/address'
+import { NavigationProp, StackActions, useNavigation } from '@react-navigation/native'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import * as React from 'react'
+import { useCallback, useState } from 'react'
+import { TouchableOpacity } from 'react-native'
+import { ScrollView } from 'react-native-gesture-handler'
+import tailwind from 'tailwind-rn'
+import { Text, TextInput, View } from '../../../../components'
+import { PrimaryColor, PrimaryColorStyle } from '../../../../constants/Theme'
+import { useTokensAPI } from '../../../../hooks/wallet/TokensAPI'
+import { BalanceParamList } from './BalancesNavigator'
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { WhaleWalletAccount } from '@defichain/whale-api-wallet'
+import { Ionicons } from '@expo/vector-icons'
+import { PrimaryButton } from '../../../../components/PrimaryButton'
+import { translate } from '../../../../translations'
+import { getWhaleClient } from '../../../../middlewares/api/whale'
+import { getDefaultWallet } from '../../../../middlewares/wallet'
+
+export type ConversionMode = 'utxosToAccount' | 'accountToUtxos'
+type Props = StackScreenProps<BalanceParamList, 'ConvertScreen'>
+
+interface ConversionIO extends AddressToken {
+  unit: 'UTXOS' | 'Token'
+}
+
+export function ConvertScreen (props: Props): JSX.Element {
+  const navigation = useNavigation<NavigationProp<BalanceParamList>>()
+
+  const tokens = useTokensAPI()
+  const [sourceToken, targetToken] = getDFIBalances(props.route.params.mode, tokens)
+
+  const [amount, setAmount] = useState<string>('0')
+  const convAmount = new BigNumber(amount).isNaN() ? '0' : new BigNumber(amount).toString()
+  const resultBal = new BigNumber(targetToken.amount).plus(convAmount)
+  const account = getDefaultWallet().get(0)
+  const whaleApiClient = getWhaleClient()
+
+  const convert = useCallback(() => {
+    constructSignedConversionAndSend(
+      whaleApiClient,
+      account,
+      props.route.params.mode,
+      new BigNumber(amount)
+    ).then(() => {
+      navigation.dispatch(StackActions.popToTop())
+    }).catch(e => console.log(e)) // TODO: display error
+  }, [amount])
+
+  return (
+    <View style={tailwind('w-full h-full')}>
```

Comment:
With this current setup, I think we're not providing option to swap modes. It would be better to have that on this page since it will be very difficult to change conversion modes. (Go back to Balances Page, click another token, Click convert)

Reply:
Yup, not a changeable state.

Reply:
But it should right?

Reply:
Oh I misunderstood you (improvised the design with simplicity). Yes, allowing change mode reduce unnecessary navigation. Reworking the UI based on latest design today.